src/shogun/regression/GLM.cpp

```diff
@@ -7,18 +7,32 @@
 #include <utility>
 using namespace shogun;
 
-GLM::GLM(): LinearMachine(){
-    init();
+GLM::GLM() : LinearMachine()
+{
+	init();
 }
 
-void GLM::init(){
-
+void GLM::init()
+{
+	SG_ADD(&m_alpha, "alpha", "alpha parameter", ParameterProperties::HYPER);
+	SG_ADD(&m_lambda, "lambda", "lambda parameter", ParameterProperties::HYPER);
+	SG_ADD(
+	    (std::shared_ptr<SGObject>*)&m_descend_updater, "descend_updater",
+	    "Descend Updater used for updating weights",
+	    ParameterProperties::SETTING);
+	SG_ADD(&m_family, "family", "family used", ParameterProperties::SETTING);
+	SG_ADD(
+	    &m_linkfn, "linkfn", "Link function used",
+	    ParameterProperties::SETTING);
 }
-GLM::GLM(DescendUpdaterWithCorrection* descend_updater, Family family, LinkFunction Link_fn, 
-	float64_t alpha, float64_t lambda): LinearMachine(){
-        m_alpha= alpha;
-        m_lambda= lambda;
-        m_linkfn= Link_fn;
-        m_descend_updater = descend_updater;
-        m_family= family;
-    }
\ No newline at end of file
+GLM::GLM(
+    std::shared_ptr<DescendUpdater> descend_updater, Family family,
```

Comment:
`const std::shared_ptr<DescendUpdater>& descend_updater`

Reply:
funny it built the first time, changed it tho